src/ddmd/builtin.d

```diff
@@ -216,6 +216,37 @@ extern (C++) Expression eval_yl2xp1(Loc loc, FuncDeclaration fd, Expressions* ar
     return new RealExp(loc, result, arg0.type);
 }
 
+extern (C++) Expression builtin_ctfeWrite(Loc loc, FuncDeclaration fd, Expressions* arguments)
+{
+    import core.stdc.stdio : fprintf;
+    import ddmd.ctfeexpr : resolveSlice;
+
+    Expression arg0 = (*arguments)[0];
+    StringExp se;
+
+    switch (arg0.op)
+    {
+        case TOKstring:
+            se = cast(StringExp) arg0;
+            break;
+        case TOKslice:
+            se = resolveSlice(arg0).toStringExp();
```

Comment:
AFAIU, if it doesn't succeed, then `toStringExp()` return `null`, i.e. `se` will remain `null` and the check in line 242 won't succeed. If you are referring to `resolveSlice(arg0)`, AFAIU from its body, it can't return `null`, so no `null` deref at the dot.